lib/src/components/NxCombobox/NxCombobox.tsx

```diff
@@ -375,6 +378,11 @@ function NxComboboxRender<T extends string | number = string>(
   );
 }
 
-const NxCombobox = Object.assign(forwardRef(NxComboboxRender), { propTypes });
+const NxCombobox = Object.assign(forwardRef(NxComboboxRender), {
+  // propTypes types can actually effect the overall type of NxCombobox in ways we don't want, the cast
+  // to `any` prevents that
+  /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+  propTypes: propTypes as any
```

Comment:
> propTypes types can actually effect the overall type of NxCombobox in ways we don't want

Out of curiosity, is it because the conditional type is added for NxCombobox?

Reply:
> Out of curiosity, is it because the conditional type is added for NxCombobox?

That might be what's tripping it up but it's something from the React typings that I don't fully understand.  I just know I've seen times where attaching the `propTypes` to a component changes the TS type for that component in a way that doesn't always work if the props or component have generic type parameters.  We do want to ensure that our PropTypes match our TS interfaces (which is why we use the `ValidationMap` type where the `propTypes` are defined), but we aren't relying on PropTypes _in lieu_ of TS interfaces, so we really don't want/need them to affect the component type like that.

Reply:
Thanks for the explanations, it is helpful.